lib/block_view_types.go

```diff
@@ -1324,6 +1339,17 @@ func (op *UtxoOperation) RawEncodeWithoutMetadata(blockHeight uint64, skipMetada
 		}
 	}
 
+	// StateChangeMetadata
+	// FIXME: This hackIsRunningStateSyncer() call is a hack to get around the fact that
+	// we don't have a way to not require a resync while introducing the state change
+	// metadata to the utxo operation struct. We don't want to use a block height to gate
+	// this because we want to be able to get state change metadata for ALL transactions.
+	// We should replace this with a more elegant solution, a better hack, or bundle it
+	// in with a release that requires a resync anyway.
+	if hackIsRunningStateSyncer() && op.StateChangeMetadata != nil {
+		data = append(data, EncodeToBytes(blockHeight, op.StateChangeMetadata, skipMetadata...)...)
+	}
```

Comment:
This had to be moved above the pos migration. Otherwise, we would require a resync which is not ideal.